### PR TITLE
HYDRA-1120 : Support compilation with usd 24.08

### PR DESCRIPTION
**Pull request by lanierd-adsk**

### CODE REVIEW

**Comment by lilike-adsk** [comment]
@lanierd-adsk Seems there's a build error, we need to update ecg-maya-hydra/build.py like https://git.autodesk.com/media-and-entertainment/ecg-usd-build/blob/master/build.py#L208 first.

11:48:54      CMake Error at CMakeLists.txt:16 (cmake_minimum_required):
11:48:54        CMake 3.27.7 or higher is required.  You are running version 3.20.3

**Comment by lilike-adsk** [comment]
or better to seperate the CMake version change to another PR?

**Comment by lanierd-adsk** [comment]
FYI : @ppt-adsk @debloip-adsk

File: CMakeLists.txt [hunk 1]
```diff
@@ -13,7 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-cmake_minimum_required(VERSION 3.13...3.17)
+cmake_minimum_required(VERSION 3.13...3.30)
```
Comment (lilike-adsk) [inline CMakeLists.txt:16]: Note that, this still needs to be changed to 3.27.7...3.30 to align with build.md, we can do it in another PR.

**Review by lilike-adsk — APPROVED** [review]

File: lib/adskHydraSceneBrowser/lib/CMakeLists.txt [hunk 2]
```diff
@@ -140,10 +153,17 @@ target_include_directories(${TARGET_NAME}
...
+	target_compile_definitions(${TARGET_NAME} 
+		PRIVATE 
+			HDUI_EXPORT
+	)
+else()
+	target_compile_definitions(${TARGET_NAME} 
+		PRIVATE 
+			HDUI_EXPORTS
+	)
+endif()
```
Comment (ppt-adsk) [inline lib/adskHydraSceneBrowser/lib/CMakeLists.txt:166]: Cleaner to use a generator expression, e.g.
https://github.com/Autodesk/maya-hydra/blob/2a49b976652f2f90eaa00fcd3e4fb0d8285c01da/cmake/compiler_config.cmake#L52

Comment (lanierd-adsk) [reply lib/adskHydraSceneBrowser/lib/CMakeLists.txt:166]: Thanks, I'll do it in my next PR as this one has already been merged. The modified code which works fine with usd 24.08 and 24.05 is :
target_compile_definitions(${TARGET_NAME} 
	PRIVATE 
	$<IF:$<VERSION_LESS:${PXR_VERSION},2408>,HDUI_EXPORT,HDUI_EXPORTS>
)
